components/views/dropdown-menu-view.js

```diff
@@ -0,0 +1,262 @@
+                'click .js-dropdown-button': 'clickOpenDropdown',
+                'click a': 'analyticsLinkClick',
+                'keydown': 'viewKeypress'
+            },
+
+            menu: '.dropdown-menu',
+
+            key: {
+                tab: 9,
+                enter: 13,
+                esc: 27,
+                up: 38,
+                down: 40
+            },
+
+            initialize: function( options ) {
+                this.$parent = $(options.parent);
+                this.render();
+            },
+
+            className: function() {
+                return this.options.className;
+            },
+
+            render: function() {
+                this.$el.html( this.tpl( this.model.toJSON() ) );
+                this.$parent.replaceWith( this.$el );
+                this.postRender();
+
+                return this;
+            },
+
+            postRender: function() {
+                this.$menu = this.$el.find('.dropdown-menu');
+                this.$page = $(document);
+                this.$dropdownButton = $('.js-dropdown-button');
+                this.listenForPageKeypress();
+            },
+
+            analyticsLinkClick: function( event ) {
+                /**
+                 *  Add your own analytics tracking here
+                 *  for example:
+                 *
+                 *  var $link = $(event.target),
+                 *      label = $link.hasClass('user-title') ? 'Dashboard' : $link.html().trim();
+                 *
+                 *  window.analytics.track( 'user_dropdown.clicked', {
+                 *      category: 'navigation',
+                 *      label: label,
+                 *      link: $link.attr('href')
+                 *  });
+                 */
+
+                return event;
+            },
+
+            clickCloseDropdown: function( event, context ) {
+                var $el = $(event.target);
+
+                if ( !$el.hasClass('button-more') && !$el.hasClass('has-dropdown') ) {
+                    context.closeDropdownMenus();
+                }
+            },
+
+            clickOpenDropdown: function( event ) {
+                event.preventDefault();
+
+                this.openMenu( $(event.target) );
+            },
+
+            closeDropdownMenus: function( all ) {
+                var $open;
+
+                if ( all ) {
+                    // Close all open, usually from ESC or doc click
+                    $open = this.$page.find( this.menu );
+                } else {
+                    // Closing one for another
+                    $open = this.$page.find( this.menu ).not( ':focus' );
+                }
+
+                $open.removeClass( 'is-visible' )
+                     .addClass( 'is-hidden' );
+
+                this.$dropdownButton
+                    .removeClass( 'is-active' )
+                    .attr( 'aria-expanded', 'false' );
+            },
+
+            escKeypressHandler: function( event ) {
```

Comment:
I think the actual JS event is called 'keypress' (https://developer.mozilla.org/en-US/docs/Web/Events/keypress) and the jQuery function is one word too.